src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -167,6 +202,14 @@ private[impl] class InstanceTrackerActor(
     }
   }
 
+  private def updateRepository(repositoryFunc: () => Future[Done], ack: Ack)(implicit ec: ExecutionContext): Future[RepositoryUpdateResult] = {
+    repositoryFunc()
+      .map(_ => RepositoryStateUpdated(ack))
+      .recoverWith {
+        case e => Future.successful(RepositoryStateUpdateFailed(e))
```

Comment:
I get why are you doing it but throwing a `Future.successful` on a failed operation looks strange 🤔

Reply:
yeah, I was thinking about this one too... I did it this way just so I can wrap that thing before using pipeTo :/ I think the whole way we use ask pattern all over place is broken.

Reply:
It seems like repository func is unneeded? It is called once and called immediately. Better to just pass in the Future?